Tests

somexpert opened the pull request:
This branch deals with everything being done to get this django app ready for a github release. So it includes documentation, a requirements file, CI integration, etc., but the really important additions are the tests to make sure we know what we need to do to make this work as expected, and to make sure it works with Django 1.8 and Python 2.7.

Commit:
Put the overall directory structure in place. Created the django project for development and running tests.

Commit:
Added README, CI integration, MANIFEST, requirements, and setup.py. Also set out some of the testing framework.

Commit:
Added more tests.

somexpert commented:
@ldko, @damonkelley, I'm kind of stuck. Do you guys have any suggestions on extra tests, or changes to the tests I've already added?

somexpert commented:
@ldko, @damonkelley, I'm kind of stuck. Do you guys have any suggestions on extra tests, or changes to the tests I've already added?

Commit:
Removed empty tests and simplified some tests.

Commit:
Replaced tab char with spaces in .travis.yml

Commit:
Modified README and changed version to 1.0.0 in setup.py

Commit:
Fixed error in command to install django-edtf from source.

Commit:
Removed unused import.

somexpert commented:
@ldko, @damonkelley, this PR is ready for review.

damonkelley commented on .travis.yml:
```diff
@@ -0,0 +1,9 @@
+sudo: false
+python: 2.7
+env:
+        - TOX_ENV=py27-django18
```
There is a little bit extra whitespace here

damonkelley commented on README.md:
```diff
@@ -0,0 +1,58 @@
+Installation
+------------
+
+1. Download and install from source code.
```
I don't think "from source code" is what we want here.

damonkelley commented on tests/urls.py:
```diff
@@ -0,0 +1,22 @@
+Including another URLconf
+    1. Add an import:  from blog import urls as blog_urls
+    2. Add a URL to urlpatterns:  url(r'^blog/', include(blog_urls))
+"""
```
Can we remove this docstring?

Commit:
Revised some README wording, removed extra whitespace and fixed env name in .travis.yml, removed docstring in urls.py, and removed django18 from testenv name in tox.

damonkelley commented:
Is `requirements.txt` meant to be the "dev requirements"? If so, we should add `edtf-validate`

Commit:
Added edtf-validate to requirements.txt.

somexpert commented:
Yep, you are correct. Adding edtf-validate.

damonkelley commented on tests/settings.py:
```diff
@@ -0,0 +1,64 @@
+
+DEBUG = True
+
+ALLOWED_HOSTS = []
```
We can remove this.

damonkelley commented on tests/settings.py:
```diff
@@ -0,0 +1,64 @@
+
+MAINTENANCE_MSG = ''
+
+STATIC_URL = '/static/'
```
We can also remove the duplicate `STATIC_URL` and `MAINTENANCE_MSG`

damonkelley replied:
Scratch the `MAINTENANCE_MSG`. The app is looking it up somewhere. We will want to make an issue to remove that.

Commit:
Removed unneeded and redundant settings in settings.py

damonkelley commented:
:+1:

ldko commented on tests/tests.py:
```diff
@@ -0,0 +1,97 @@
+
+    # The view should return an HTTP 400 response when no level is present.
+    @expectedFailure
+    def test_returns_400_without_level(self):
```
The way I read boolean_result(), it should return a 200 if there is a date but no level. level is not required. This view is weird cause it isn't actually used and if theoretically you do submit with a level, this view doesn't make much use of it.

ldko commented on tests/tests.py:
```diff
@@ -0,0 +1,97 @@
+    def setUp(self):
+        self.factory = RequestFactory()
+
+    def test_returns_200(self):
```
boolean_result() implies that if the `date` value is an empty string it should give a 400. Because the code is written `date is ''` instead of `date == ''` making a request with date='' is giving a 200 when it should probably be 400. In that case, I would say the expected status_code for this test should be 400 and that this should be expected to fail because of a bug in the view code. That being said, boolean_result does not seem to be used, seems a bit broken, and we could probably skip testing it, remove the view code and the commented out urls.py stuff related to it (and while we are at it, remove the urlerror() function in views.py that is also apparently not used).

somexpert replied:
I would _love_ to get rid of those views. And yeah, I didn't put too much work into the tests for the views since I thought it was very likely they might go away. They were pretty much copy-pasted from the result_json tests. I'll delete the mentioned views and url stuff, as well as these tests.

ldko commented on tests/tests.py:
```diff
@@ -0,0 +1,97 @@
+    def setUp(self):
+        self.factory = RequestFactory()
+
+    def test_returns_200(self):
```
`result_json` implies that if the date value is an empty string it should give a 400. Because the code is written date is '' instead of date == '' making a request with date='' is giving a 200 when it should probably be 400.

ldko commented on tests/tests.py:
```diff
@@ -0,0 +1,97 @@
+    def test_returns_400_without_level(self):
+        request = self.factory.get('/', {'date': 'March 3rd, 2016', 'level': ''})
+        response = result_json(request)
+        self.assertEqual(response.status_code, 400)
```
It would be good for there to be additional testing of result_json for values of `level` other than ''.

Commit:
Added tests and removed unneeded urls and views.

Added tests to cover more of the result_json view, as well as removing
the url_error url and view, and the boolean_result url and view.

ldko commented:
:+1:

ldko commented:
errr actually there is some pep8 stuff in multiple files that should be cleaned up before merging. Then merge! Thanks!

Commit:
Fixed PEP8 stuff.
